internal/peer/peers_service.go

```diff
@@ -0,0 +1,43 @@
+package peer
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/facebookgo/startstop"
+)
+
+// Peers provides information about all peers in a Refinery cluster.
+type Peers interface {
+	Subscribe(func(PeerInfo))
```

Comment:
Why callbacks instead of channels?

Also, please add documentation in the interfaces used by external clients. That's what shows up in hovers in the editor, so named parameters and comments are better put here instead of (in addition to) the implementation.

Reply:
I added the comments.
I personally find the callback pattern more readable but I can go either way. It feels like it is less change to make to keep the channel pattern so I made the change